test/lit/basic/export-import.wast

```diff
@@ -14,21 +14,26 @@
  ;; CHECK-BIN:      (type $v (func))
  (type $v (func))
  ;; CHECK-TEXT:      (import "env" "test2" (global $test2 i32))
- ;; CHECK-BIN:      (import "env" "test2" (global $test2 i32))
- ;; CHECK-BIN-NODEBUG:      (type $0 (func))
 
- ;; CHECK-BIN-NODEBUG:      (import "env" "test2" (global $gimport$0 i32))
- (import "env" "test1" (func $test1))
  ;; CHECK-TEXT:      (import "env" "test1" (func $test1 (type $v)))
+ ;; CHECK-BIN:      (import "env" "test2" (global $test2 i32))
+
  ;; CHECK-BIN:      (import "env" "test1" (func $test1 (type $v)))
- ;; CHECK-BIN-NODEBUG:      (import "env" "test1" (func $fimport$0 (type $0)))
+ (import "env" "test1" (func $test1))
  (import "env" "test2" (global $test2 i32))
  ;; CHECK-TEXT:      (export "test1" (func $test1))
  ;; CHECK-BIN:      (export "test1" (func $test1))
- ;; CHECK-BIN-NODEBUG:      (export "test1" (func $fimport$0))
  (export "test1" (func $test1))
  ;; CHECK-TEXT:      (export "test2" (global $test2))
  ;; CHECK-BIN:      (export "test2" (global $test2))
- ;; CHECK-BIN-NODEBUG:      (export "test2" (global $gimport$0))
  (export "test2" (global $test2))
 )
+;; CHECK-BIN-NODEBUG:      (type $0 (func))
+
+;; CHECK-BIN-NODEBUG:      (import "env" "test2" (global $gimport$0 i32))
+
+;; CHECK-BIN-NODEBUG:      (import "env" "test1" (func $fimport$0 (type $0)))
+
+;; CHECK-BIN-NODEBUG:      (export "test1" (func $fimport$0))
+
+;; CHECK-BIN-NODEBUG:      (export "test2" (global $gimport$0))
```

Comment:
Looks less readable here, as it emits it out past the end of the module for some reason?

Reply:
This is because round-tripping through the binary does not preserve the names we are now correctly matching against. That's not a new problem with this change, although it now correctly shows up in this case.

Reply:
But before it looks like it was just in the right place? I mean, before we had this:
```wat
 ;; CHECK-BIN: (import "env" "test1" (func $test1 (type $v)))
 ;; CHECK-BIN-NODEBUG:      (import "env" "test1" (func $fimport$0 (type $0)))
```
So yeah, we didn't round-trip `test1` but still the imports were perfectly aligned?

Reply:
It was in a nice place for the wrong reasons. If the input had more imports from "env", then they would all be bunched up together in the output rather than actually matched up with their corresponding input.

Reply:
Oh, I actually think grouping all imports that are identical in module and name makes sense - do you disagree? In the quote from my last comment, `env` and `test1` are identical in both lines, so it's nice to keep them together.

I can see how a more strict approach can justify moving them away, but I think it's a loss for readability. Could we perhaps both match names more precisely and also secondarily keep imports/exports with identical external names close together? Then there would not be a regression here from my perspective.

But if that's very difficult to do I won't insist.

Reply:
There are other cases (such as in the next PR) where this improves readability by correctly matching imports with their inputs where they would not have otherwise been grouped together.

The update script never reorders the input or the output, so in that sense all the output imports will always be grouped together, but now the way they are interleaved with the input is more consistent and the result is more readable in some cases, but not all cases.

Doing better than making the interleaving consistent would require reordering things, and that's not possible or desirable right now.